gui/default/syncthing/core/syncthingController.js

```diff
@@ -3077,6 +3077,120 @@ angular.module('syncthing.core')
+            // other methods, hence we define them separately before the body.
+            var title = $translate.instant("{%devicename%} has shared their Syncthing device ID!", {devicename: deviceName});
+            var footer = $translate.instant("Learn more at {%url%}.", {url: "https://syncthing.net"});
+
+            switch (type) {
+                case "email":
+                    params.heading = $translate.instant("Share by Email");
+                    params.icon = "fa fa-envelope-o";
+                    // Email message format requires using CRLF for line breaks.
+                    // Ref: https://datatracker.ietf.org/doc/html/rfc5322
+                    params.subject = title;
+                    params.body = [
+                        $translate.instant("You have been invited to connect with {%devicename%} using Syncthing!", {devicename: deviceName}),
+                        $translate.instant("To accept the invitation, add a new device using the ID below."),
+                        deviceID,
+                        $translate.instant("Syncthing is a continuous file synchronization program. It synchronizes files between two or more computers in real time, safely protected from prying eyes. Your data is your data alone and you deserve to choose where it is stored, whether it is shared with some third party, and how it's transmitted over the internet."),
+                        footer
+                    ].join('\r\n\r\n');
+                    break;
+                case "sms":
+                    params.heading = $translate.instant("Share by SMS");
+                    params.icon = "fa fa-comments-o";
+                    // SMS is limited to 160 characters (non-Unicode), so we keep
+                    // it as short as possible, e.g. by stripping hyphens from
+                    // device ID. The current minimum length is around 140 chars,
+                    // but some room is required for longer sharing device names.
+                    params.body = [
+                        title,
+                        deviceID.replace(/-/g, ''),
+                        footer
+                    ].join('\n\n');
+                    break;
+                default:
+            }
+            $scope.shareDeviceParams = params;
+            $('#share-device-confirmation').modal('show');
+        };
+
+        $scope.shareDevice = function () {
+            switch ($scope.shareDeviceParams.type) {
+                case 'email':
+                    location.href = 'mailto:?subject=' + encodeURIComponent($scope.shareDeviceParams.subject) + '&body=' + encodeURIComponent($scope.shareDeviceParams.body);
+                    break;
+                case 'sms':
+                    location.href = 'sms://;?&body=' + encodeURIComponent($scope.shareDeviceParams.body);
+                    break;
+                default:
+            }
+        }
+
+        $scope.showTooltip = function (newTooltip) {
```

Comment:
Alerts are usually a no-go in general, mainly because they require an immediate action preventing the user from interacting with the document completely, but also they are extremely abused by spammy sites in order to prevent the user from leaving the site, etc., so many people have grown to hate them in principle. That's why after more testing I've decided that it'd be much better to use the modal notifications instead.

However, this particular case is different, as the purpose of the function is simply to either show a temporary tooltip or replace the existing one, so that the user knows that their action has been successful (or not), .e.g. `Copy to Clipboard` becomes `Copied to Clipboard!`, etc. I will add the proper explanation, so that everything should be clear about its purpose.

Reply:
That's alright. I just meant you should put a comment on the function what its purpose and mechanism is. Maybe some less generic naming would help also. Tool tips are used in many places, so this should make obvious how it's different from a regular tool tip.

Reply:
Explanation added. Please let me know if you've got a better, and maybe more specific, name in mind. The function itself can be used to add a temporary tooltip to any element though, hence I've used a more comprehensive name for it.

Reply:
`showTemporaryTooltip`?

It's not quite clear in your comment what "afterwards" refers to.  I think you mean after the temporary one is shown, the data for an event-driven tooltip is restored.

Reply:
Renamed to `afterwards` and replaced "afterwards" with a better explanation.